### PR TITLE
Home page

### PR DESCRIPTION
**Pull request by ChiragdevRobosoft**
I have made all major changes, today due to some changes made to adjust rakesh's and anusha's components with home page lead to complete change in homepage layout. All pages are now normalised except for searchBox component>

Rakesh needs the header component to complete his task hence i have completed that part and raised PR.

### CODE REVIEW

File: src/components/common/carousels/bestDealsOffers/index.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+export default CarouselBDO;
+
+const Wrapper = styled.div`
+  .main-slide {
+    height: ${sizes.size287};
+    width: ${sizes.size960};
+    margin: ${sizes.size0} ${sizes.size0} ${sizes.size0} ${sizes.size0};
+    border-radius: ${sizes.size6};
+    //background-color: ${colors.white};
+    background-color: linear-gradient(178.89deg, #ffffff 0%, #f4f6ff 100%);
```
Comment (naikashok) [inline src/components/common/carousels/bestDealsOffers/index.tsx]: Color code should move to variable file.

File: src/components/common/carousels/popularBrands/index.tsx [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+import kfcLogo from "./../../../../assets/common/carousels/brand_logo.png";
+import emptyIcon from "./../../../../assets/common/carousels/noRestaurants.jpg";
+import tData from "./../../constants.json";
+const Carousel = require("react-responsive-carousel").Carousel;
+
+let datas: string[] = ["one", "two", "three", "four"];
+
+const renderArrow =
+  (direction: string) =>
+  (onClickHandler: any, shouldBeEnabled: boolean, label: any) => {
```
Comment (naikashok) [inline src/components/common/carousels/popularBrands/index.tsx]: Use specific datatype, don;t use any.

File: src/components/common/carousels/popularBrands/index.tsx [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+    }
+
+    return <div onClick={onClickHandler} style={styles}></div>;
+  };
+
+const CarouselPB = () => {
+  return (
+    <>
+      {datas.length === 0 ? (
+        <NoData>
```
Comment (naikashok) [inline src/components/common/carousels/popularBrands/index.tsx:80]: Move all the hard-coded string to .json file

File: src/components/common/foodSearchBox/index.tsx [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+const Foodsearch: FC<foodSearchProps> = ({ pageType }) => {
+  return (
+    <Wrapper className={pageType}>
+      <SearchBox
+        className={pageType}
+        type="text"
+        name="food-search"
+        placeholder="Search your restaurant or cuisines"
+      />
+      <LocDateDiv className={pageType}>
```
Comment (naikashok) [inline src/components/common/foodSearchBox/index.tsx]: Use meaningful names instead off using "LocDateDiv", don't use div.

File: src/components/common/carousels/bestDealsOffers/index.tsx [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+  {
+    one: "./../../../../assets/common/carousels/Card1.png",
+    two: "./../../../../assets/common/carousels/Card1.png",
+  },
+  { one: "./../../../../assets/common/carousels/Card1.png" },
+];
+
+const renderArrow =
+  (direction: string) =>
+  (onClickHandler: any, shouldBeEnabled: boolean, label: any) => {
```
Comment (naikashok) [inline src/components/common/carousels/bestDealsOffers/index.tsx]: Remove any keyword and use particular data-type

File: src/components/common/carousels/bestDealsOffers/index.tsx (hunk 1 above)
Comment (ChiragdevRobosoft) [reply src/components/common/carousels/bestDealsOffers/index.tsx]: colour codes shifted into variables.tsx. Comments also removed wherever present.

File: src/components/common/carousels/popularBrands/index.tsx (hunk 2 above)
Comment (ChiragdevRobosoft) [reply src/components/common/carousels/popularBrands/index.tsx]: specific datatype given based on respective event-listener types.

File: src/components/common/carousels/popularBrands/index.tsx (hunk 3 above)
Comment (ChiragdevRobosoft) [reply src/components/common/carousels/popularBrands/index.tsx:80]: all textual data moved , only html tag names are not converted for recognition.If required i will change that also.

File: src/components/common/foodSearchBox/index.tsx (hunk 4 above)
Comment (ChiragdevRobosoft) [reply src/components/common/foodSearchBox/index.tsx]: I have changed the tag names and tried to give reasonable names.

File: src/components/common/carousels/bestDealsOffers/index.tsx (hunk 5 above)
Comment (ChiragdevRobosoft) [reply src/components/common/carousels/bestDealsOffers/index.tsx]: specific datatype given based on respective event-listener types.

**Review by naikashok — APPROVED** [review]

**Pull request merged**